web/src/components/Layout/Profile/CreatedCollectionList.tsx

```diff
@@ -0,0 +1,155 @@
+import { useMutation, useQuery } from "@tanstack/react-query";
+import { useRouter } from "next/router";
+import { useEffect, useMemo, useState } from "react";
+import { toast } from "react-hot-toast";
+
+import { getCollections } from "../../../requests/fetch";
+import { createCollection } from "../../../requests/mutate";
+import Collection from "../../../types/collection.type";
+import ProfileSearchBar from "../../Inputs/ProfileSearchBar";
+import CollectionList from "../CollectionList";
+import ImageList from "../ImageList";
+
+interface Props {
+  userId?: string;
+  isSelf: boolean;
+  isProfileLoading: boolean;
+}
+
+export default function CreatedCollectionList({
+  userId,
+  isSelf,
+  isProfileLoading,
+}: Props) {
+  //#region Hooks
+
+  const router = useRouter();
+
+  const [selectedCollection, setSelectedCollection] =
+    useState<Collection | null>(null);
+  const [isAddCollectionDisabled, setIsAddCollectionDisabled] = useState(true);
+  const [searchCollectionValue, setSearchCollectionValue] = useState("");
+  const [searchPostValue, setSearchPostValue] = useState("");
+
+  const {
+    data: collectionData,
+    isLoading: areCollectionsLoading,
+    refetch: refetchCollections,
+  } = useQuery({
+    queryKey: ["user_collections", userId],
+    queryFn: () => getCollections({ userId }),
+    onError: (err: Error) => {
+      toast.error(err.message);
+    },
+    enabled: !!userId,
+  });
+
+  const { mutate: mutateCreateCollection, isLoading: isCreatingCollection } =
+    useMutation({
+      mutationKey: ["create_collection"],
+      mutationFn: () => createCollection(searchCollectionValue.trim()),
+      onSuccess: () => {
+        void refetchCollections();
+      },
+      onError: (err: Error) => {
+        toast.error(err.message);
+      },
+    });
+
+  const collections = useMemo(
+    () =>
+      collectionData?.collections.filter((collection) =>
+        collection.name
+          .toLowerCase()
+          .includes(searchCollectionValue.toLowerCase())
+      ),
+    [collectionData, searchCollectionValue]
+  );
+
+  const selectedCollectionPosts = useMemo(
+    () =>
+      selectedCollection?.posts.filter((post) =>
+        post.prompt.toLowerCase().includes(searchPostValue.toLowerCase())
+      ),
+    [selectedCollection, searchPostValue]
+  );
+
+  useEffect(() => {
+    if (
+      searchCollectionValue.trim().length === 0 ||
+      collections?.some(
+        (collection) =>
+          collection.name.toLowerCase() ===
+          searchCollectionValue.trim().toLowerCase()
+      )
+    ) {
+      setIsAddCollectionDisabled(true);
+    } else {
+      setIsAddCollectionDisabled(false);
+    }
+  }, [collections, searchCollectionValue]);
```

Comment:
I don't think you need `useEffect()` for this logic. My rule of thumb is that any time you see `useEffect()` without a DOM side-effect, you probably don't need it.
```ts
const isAddCollectionDisabled = searchCollectionValue.trim().length === 0 ||
      collections?.some(
        (collection) =>
          collection.name.toLowerCase() ===
          searchCollectionValue.trim().toLowerCase()
      )
```
I think the above does what you want.